### PR TITLE
FEAT: Download image

### PR DESCRIPTION
**Pull request by drake-ho-goldenowl**
## Task
- [X] Update UI touch image
- [X] Download the image

### CODE REVIEW

File: app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseBottomSheetDialog.kt [hunk 1]
```diff
@@ -25,6 +28,16 @@ open class BaseBottomSheetDialog : BottomSheetDialogFragment() {
...
+            btn.visibility = View.GONE
+        } else {
+            progress.visibility = View.GONE
+            btn.visibility = View.VISIBLE
+        }
+    }
+
     companion object {
         const val GRIDVIEW_SPAN_COUNT = 3
         const val WARNING_SELECT_SIZE = "Please select size"
```
Comment (dustin-goldenowl) [inline app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseBottomSheetDialog.kt:43]: If can, put in String resource

File: app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseFragment.kt [hunk 2]
```diff
@@ -35,6 +37,14 @@ open class BaseFragment : Fragment() {
         }
     }
 
+    fun setLoadingCustom(progress: ProgressBar, isLoading: Boolean){
+        if (isLoading) {
```
Comment (dustin-goldenowl) [inline app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseFragment.kt:41]: can replace by progress.isVisible = isLoading

File: app/src/main/java/com/goldenowl/ecommerceapp/utilities/Constants.kt [hunk 3]
```diff
@@ -56,6 +56,7 @@ const val DELIVERED = 0
 const val PROCESSING = 1
 const val CANCELLED = 2
 
+const val SUCCESS_DOWNLOAD = "Saved success"
 const val WARNING_CHECK_AGAIN = "Please check again"
```
Comment (dustin-goldenowl) [inline app/src/main/java/com/goldenowl/ecommerceapp/utilities/Constants.kt:60]: put in string source

File: app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseBottomSheetDialog.kt (hunk 1 above)
Comment (drake-ho-goldenowl) [reply app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseBottomSheetDialog.kt:43]: done

File: app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseFragment.kt (hunk 2 above)
Comment (drake-ho-goldenowl) [reply app/src/main/java/com/goldenowl/ecommerceapp/ui/BaseFragment.kt:41]: done

File: app/src/main/java/com/goldenowl/ecommerceapp/utilities/Constants.kt (hunk 3 above)
Comment (drake-ho-goldenowl) [reply app/src/main/java/com/goldenowl/ecommerceapp/utilities/Constants.kt:60]: done

**Pull request merged**